subsys/nv_config/src/config.c

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include <errno.h>
+
+#include "mbedtls/base64.h"
+
+#include "config/config.h"
+#include "config_priv.h"
+#include <zephyr/types.h>
+
+/* mbedtls-base64 lib encodes data to null-terminated string */
+#define BASE64_ENCODE_SIZE(in_size) ((((((in_size) - 1) / 3) * 4) + 4) + 1)
+
+struct conf_handler_head conf_handlers;
+
+static u8_t conf_cmd_inited;
+
+void conf_init(void)
+{
+	int rc;
+
+	SLIST_INIT(&conf_handlers);
+	conf_store_init();
+
+	if (conf_cmd_inited) {
+		return;
+	}
+
+	(void)rc;
+
+#ifdef CONFIG_NV_CONFIG_SYSTEM_CLI
+	rc = conf_cli_register();
+	SYSINIT_PANIC_ASSERT(rc == 0);
+#endif
+#ifdef CONFIG_NV_CONFIG_SYSTEM_MCUMGR
+	rc = conf_nmgr_register();
+	SYSINIT_PANIC_ASSERT(rc == 0);
+#endif
+
+	conf_cmd_inited = 1;
+}
+
+int conf_register(struct conf_handler *handler)
+{
+	SLIST_INSERT_HEAD(&conf_handlers, handler, ch_list);
+	return 0;
+}
+
+/*
+ * Find conf_handler based on name.
+ */
+struct conf_handler *conf_handler_lookup(char *name)
+{
+	struct conf_handler *ch;
+
+	SLIST_FOREACH(ch, &conf_handlers, ch_list) {
+		if (!strcmp(name, ch->ch_name)) {
+			return ch;
+		}
+	}
+	return NULL;
+}
+
+/*
+ * Separate string into argv array.
+ */
+int conf_parse_name(char *name, int *name_argc, char *name_argv[])
+{
+	char *tok;
+	char *tok_ptr;
+	char *sep = CONF_NAME_SEPARATOR;
+	int i;
+
+	tok = strtok_r(name, sep, &tok_ptr);
```

Comment:
strtok_r() isn't available in the minimal libc. Will this be an issue?

Reply:
new lib c is selected once this module is enabled.